synapseclient/models/evaluation.py

```diff
@@ -1 +1,838 @@
+        ```
+        """
+        from synapseclient.api.evaluation_services import get_evaluation_async
+
+        if not self.id and not self.name:
+            raise ValueError("Either id or name must be set to get an evaluation")
+
+        retrieved_evaluation = await get_evaluation_async(
+            evaluation_id=self.id,
+            name=self.name,
+            synapse_client=synapse_client,
+        )
+
+        self.fill_from_dict(retrieved_evaluation)
+
+        # Save the current state to track future changes
+        self._set_last_persistent_instance()
+
+        return self
+
+    async def delete_async(self, *, synapse_client: Optional["Synapse"] = None) -> None:
+        """
+        Delete this Evaluation from Synapse. ID must be set in order to delete the Evaluation.
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by `Synapse.allow_client_caching(False)` this will use the last created
+                            instance from the Synapse class constructor.
+
+        Raises:
+            ValueError: If evaluation_id is not set.
+            SynapseHTTPError: If the service rejects the request or an HTTP error occurs.
+
+        Example: Using this function
+        &nbsp;
+
+        ```python
+        from synapseclient.models import Evaluation
+        from synapseclient import Synapse
+        import asyncio
+
+        syn = Synapse()
+        syn.login()
+
+        async def delete_evaluations():
+            # Delete an evaluation by ID
+            await Evaluation(id="9614112").delete_async()
+
+            # Get and then delete an evaluation
+            # First get the evaluation by name, so the ID attribute is set in your
+            # Evaluation object, then delete it.
+            evaluation = await Evaluation(name="My Challenge Evaluation").get_async()
+            await evaluation.delete_async()
+
+        asyncio.run(delete_evaluations())
+        ```
+        """
+        from synapseclient.api.evaluation_services import delete_evaluation_async
+
+        if not self.id:
+            raise ValueError("id must be set to delete an evaluation")
+
+        await delete_evaluation_async(
+            evaluation_id=self.id,
+            synapse_client=synapse_client,
+        )
+
+        # Clear the persistent instance since this object has been deleted
+        self._last_persistent_instance = None
+
+    async def get_acl_async(
+        self, *, synapse_client: Optional["Synapse"] = None
+    ) -> dict:
+        """
+        Get the access control list (ACL) governing this evaluation.
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by `Synapse.allow_client_caching(False)` this will use the last created
+                            instance from the Synapse class constructor.
+
+        Returns:
+            AccessControlList: The ACL for this Evaluation.
```

Comment:
Right now it looks like this returns a dictionary representing the `AccessControlList` rather than an `AccessControlList` object. Once you define an AccessControlList class, you could return that instead.